enhancements/insights/config-api.md

```diff
@@ -0,0 +1,296 @@
+	// ReportStatus provides general Insights analysis results.
+	ReportStatus ReportStatus `json:"insightsReport,omitempty"`
+}
+
+type Gathering struct {
+	// Interval defines the time frequency of gathering. The duration is 2h by default.
+	// +kubebuilder:default=2h
+	Interval time.Duration `json:"interval,omitempty"`
+	// UploadEndpoint defines ingress endpoint for uploading Insights archive
+	// +kubebuilder:default=https://cloud.redhat.com/api/ingress/v1/upload
+	UploadEndpoint string `json:"endpoint,omitempty"`
+	// EnableObfuscation is flag enabling the global obfuscation of the IP addresses and base domain
+	// in the Insights archive data.
+	EnableObfuscation bool `json:"enableObfuscation,omitempty"`
+	// Enables user to force Insights data gathering by setting a new reason
+	ForceGatheringReason string `json:"forceGatheringReason,omitempty"`
+	// List of gatherers to be included in the gathering. Empty list means that all gatherers are included
+	Gatherers []string `json:"gatherers,omitempty"`
+	// InsightsReport includes all the configuration options related to downloading the Insights report.
+	InsightsReport InsightsReport `json:"insightsReport,omitempty"`
+	// Flag to disable gathering. When the gathering is disabled then it cannot be forced with the `ForceGatheringReason`
+	Disabled bool `json:"disabled,omitempty"`
+	// Path in the container for temporary storage of the Insights archive
+	// +kubebuilder:default=/var/lib/insights-operator
+	StoragePath string `json:"storagePath,omitempty"`
+}
+
+type SimpleContentAccess struct {
+	// Endpoint is the HTTP endpoint to pull down the SCA certificates from. This is set to
+	// corresponding OCM API by default.
+	// +kubebuilder:default=https://api.openshift.com/api/accounts_mgmt/v1/certificates
+	Endpoint string `json:"endpoint,omitempty"`
+	// Interval defines the time frequency of downloading the SCA certificates.
+	// The duration is 8h by default.
+	// +kubebuilder:default=8h
+	Interval time.Duration `json:"interval,omitempty"`
+	// Disabled flag is an option how to disable the SCA download feature.
+	Disabled bool `json:"disabled,omitempty"`
+}
+
+type ClusterTransfer struct {
+	// Endpoint is the HTTP endpoint to query the cluster transfer
+	Endpoint string `json:"endpoint,omitempty"`
+	// Interval defines the time frequency of checking the cluster transfer.
+	// The duration is 24h by default.
+	// +kubebuilder:default=24h
+	Interval time.Duration `json:"interval,omitempty"`
+}
+
+type InsightsReport struct {
+	// Endpoint is the HTTP endpoint to download the Insights report from.
+	// +kubebuilder:default=https://cloud.redhat.com/api/insights-results-aggregator/v1/clusters/%s/report
+	Endpoint string `json:"endpoint,omitempty"`
+	// Delay after uploading the Insights archive, when the first attempt to download the corresponding Insights report will take place.
+	// +kubebuilder:default=1m
+	Delay time.Duration `json:"delay,omitempty"`
```

Comment:
`InitialDownloadDelaySeconds` or something similar :-)

Reply:
Same as below. I would remove it from the config

Reply:
OK let's keep it for now. I changed it to `DownloadDelaySeconds int`